deepchem/models/tests/test_layers.py

```diff
@@ -1544,3 +1544,28 @@ def test_torch_graph_gather():
         np.load("deepchem/models/tests/assets/graphgatherlayer_result.npy"),
         atol=1e-4)
     assert result.shape == (batch_size, 2 * n_features)
+
+
+@pytest.mark.torch
+def test_torch_cosine_dist():
+    """Test invoking cosine_dist."""
+
+    x = torch.ones((5, 4), dtype=torch.float32)
+    y_same = torch.ones((5, 4), dtype=torch.float32)
+    # x and y are the same tensor (equivalent at every element)
+    # the pairwise inner product of the rows in x and y will always be 1
+    # the output tensor will be of shape (5,5)
+    cos_sim_same = torch_layers.cosine_dist(x, y_same)
+    diff = cos_sim_same - torch.ones((5, 5), dtype=torch.float32)
+    assert torch.abs(torch.sum(diff)) < 1e-5  # True
+
+    identity_tensor = torch.eye(
+        512, dtype=torch.float32)  # identity matrix of shape (512,512)
```

Comment:
@Dragonwagon18 Let's make the sizes of tensors small unless they're required to be this big. We're trying to make the CI leaner.